packages/report-server/src/reportBuilder/transform/functions/aggregate/aggregations.ts

```diff
@@ -3,75 +3,124 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Row, FieldValue } from '../../../types';
+import { FieldValue } from '../../../types';
 
 const isUndefined = (value: FieldValue): value is undefined => {
   return value === undefined;
 };
 
-const group = (existingRow: Row, field: string, value: FieldValue) => {
-  existingRow[field] = value;
+const filterUndefinedAndNull = (values: FieldValue[]): NonNullable<FieldValue[]> => {
+  const filteredValues: NonNullable<FieldValue[]> = [];
+  values.forEach(value => {
+    if (value !== undefined && value !== null) {
+      filteredValues.push(value);
+    }
+  });
+  return filteredValues;
+};
+
+const checkIsNum = (values: FieldValue[]): number[] => {
+  const filteredUndefinedAndNullValues = filterUndefinedAndNull(values);
+  const checkedValues: number[] = [];
+  filteredUndefinedAndNullValues.forEach(value => {
+    if (typeof value !== 'number') {
+      throw new Error(`Expected number, got '${typeof value}'.`);
+    }
+    checkedValues.push(value);
+  });
+  return checkedValues;
 };
 
-const sum = (existingRow: Row, field: string, value: FieldValue) => {
-  if (typeof value === 'number') {
-    existingRow[field] = ((existingRow[field] as number) || 0) + value;
-  } else {
-    throw new Error(`Expected number, got '${typeof value}'.`);
+const group = (values: FieldValue[]): FieldValue => {
```

Comment:
I think it's actually true that we want to filtering the values for all null/undefined for all aggregations (not just sum, etc.). I hadn't implemented this originally and had just been fixing it on an adhoc basis, but now that someone's taken the time to re-write this it would be nice to prevent future bugs. 👍

Reply:
To me filtering the null/undefined value is to prevent the calculation error, I didn't do it on `group` aggregation is because we might want to send `null` as value in some special cases. 
What I think is we can filter the `null` by `filter` transform, but you can show them if you want. With that we can provide more freedom.

Reply:
Sorry I just notice I filter undefined for every aggregations... But I still think not to filter them out unless we do calculation. @rohan-bes What do you think?

Reply:
>To me filtering the null/undefined value is to prevent the calculation error, I didn't do it on group aggregation is because we might want to send null as value in some special cases.
>
>I think we can filter the null by filter transform, but you can show them if you want. With that we can provide more freedom.

I think you make a really excellent point here! I think you have totally convinced me 🎉 you are correct that it's a potential valid case that we may want to `group` all rows with `null` for a specific field together (and not just `group`, also: `last`, `unique`, etc.)!

How does this sound:
- Always filter out `undefined`. `undefined` is fairly consistently treated as that property not existing on the row, both in javascript as a whole **_AND_**  in BES
- Keep `null`, but filter when doing mathematical calculations as you have done. `null` is treated as 'We know we have no data for this field in this row'. It's a slight distinction, but may be important, and shares consistency with the new 'fill missing values with `null`' logic you implemented in `aggregator`

Reply:
Yeah I agree we keep that consistency, and as I remember you already confirmed this logic with the teams, and the logic you commented above looks good to me 👍  so I will apply this to the code.